# HG changeset patch
# User Pieter Eendebak <pieter.eendebak@gmail.com>
# Date 1249536773 -7200
#      Thu Aug 06 07:32:53 2009 +0200
# Branch release-3-2-x
# Node ID e868e6276b5ca2b5ddaf8dd50f60b56bc07ce78f
# Parent  1740ac0eb9c28fceffec318c04c6e6bb0c2bdcc5
support cellstrs in setxor

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-04  Pieter Eendebak <pieter.eendebak@gmail.com> 
+
+	* set/setxor.m: Support cell arrays of strings.
+
 2009-08-05  Olli Saarela  <olli.saarela@gmail.com>
 
 	* plot/__gnuplot_get_var__.m: If read fails to return data, sleep
 	before trying again.
 
 2009-08-02  Ben Abbott <bpabbott@mac.com>
 
 	* plot/gnuplot_drawnow.m: Avoid the flickering x11 window seen with
diff --git a/scripts/set/setxor.m b/scripts/set/setxor.m
--- a/scripts/set/setxor.m
+++ b/scripts/set/setxor.m
@@ -66,17 +66,21 @@ function [c, ia, ib] = setxor (a, b, var
       if (! isempty (idx))
 	c([idx, idx+1],:) = [];
 	i([idx, idx+1],:) = [];
       endif
     else
       na = numel (a); nb = numel (b);
       [c, i] = sort ([a(:); b(:)]);
       n = length (c);
-      idx = find (c(1:n-1) == c(2:n));
+      if (iscell (c))
+        idx = find (strcmp (c(1:n-1), c(2:n)));	  
+      else
+        idx = find (c(1:n-1) == c(2:n));
+      endif
       if (! isempty (idx))
 	c([idx, idx+1]) = [];
 	i([idx, idx+1]) = [];
       endif
       if (size (a, 1) == 1 || size (b, 1) == 1)
 	c = c.';
       endif
     endif
